File: KMIT/BitManipulation_and_Greedy/Problem9.java
/*
Question 9:

The Hamming distance between two integers is the number of positions at which the corresponding bits are different.
Given two integers x and y, return the Hamming distance between them.


Example 1:
Input: x = 1, y = 4
Output: 2
Explanation:
1   (0 0 0 1)
4   (0 1 0 0)
       ↑   ↑
The above arrows point to positions where the corresponding bits are different.


Example 2:
Input: x = 3, y = 1
Output: 1

*/


import java.util.*;
public class Problem9 {
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        int b = sc.nextInt();
        System.out.println(hammingDist(a, b));
        sc.close();
    }
    public static int hammingDist(int a, int b) {
        int xor = a ^ b;
        int count = 0;
        while (xor > 0) {
            count += xor & 1;
            xor >>= 1;
        }
        return count;
    }
}

File: student_Banasthali_Nivedita/bitManipulation_and_Greedy/problem9.cpp
/*
Hamming distance
*/

#include<bits/stdc++.h>
using namespace std;

int hammingDistance(int x, int y) {
    int distance = 0;
    int xorValue = x ^ y;

    while (xorValue > 0) {
        if (xorValue & 1) {
            distance++;
        }
        xorValue >>= 1;
    }

    return distance;
}

int main() {
    int x, y;

    // Example 1
    x = 1;
    y = 4;
    int result1 = hammingDistance(x, y);
    cout << "Example 1: " << result1 << endl;

    return 0;
}

